monai/apps/detection/transforms/array.py

```diff
@@ -122,7 +129,148 @@ def __call__(self, boxes: NdarrayOrTensor) -> NdarrayOrTensor:
+        invert_affine: whether to apply inverse affine matrix
+    """
+
+    backend = [TransformBackends.TORCH, TransformBackends.NUMPY]
+
+    def __init__(self, invert_affine: bool = False) -> None:
+        self.invert_affine = invert_affine
+
+    def __call__(self, boxes: NdarrayOrTensor, affine: Union[NdarrayOrTensor, None]) -> NdarrayOrTensor:  # type: ignore
+        """
+        Args:
+            boxes: source bounding boxes, Nx4 or Nx6 torch tensor or ndarray. The box mode is assumed to be ``StandardMode``
+            affine: affine matric to be applied to the box coordinate
+        """
+
+        # convert boxes to torch tensor
+        if affine is None:
+            return boxes
+
+        affine_t, *_ = convert_data_type(affine, torch.Tensor)
+        if self.invert_affine:
+            affine_t = torch.inverse(affine_t)
+
+        return apply_affine_to_boxes(boxes, affine=affine_t)
+
+
+class ZoomBox(Transform):
+    """
+    Zooms an ND Box with same padding or slicing setting with Zoom().
+
+    Args:
+        zoom: The zoom factor along the spatial axes.
+            If a float, zoom is the same for each spatial axis.
+            If a sequence, zoom should contain one value for each spatial axis.
+        keep_size: Should keep original size (padding/slicing if needed), default is True.
+        kwargs: other arguments for the `np.pad` or `torch.pad` function.
+            note that `np.pad` treats channel dimension as the first dimension.
+    """
+
+    backend = [TransformBackends.TORCH, TransformBackends.NUMPY]
+
+    def __init__(self, zoom: Union[Sequence[float], float], keep_size: bool = False, **kwargs) -> None:
+        self.zoom = zoom
+        self.keep_size = keep_size
+        self.kwargs = kwargs
+
+    def __call__(
+        self, boxes: NdarrayOrTensor, src_spatial_size: Union[Sequence[int], int, None] = None
+    ) -> NdarrayOrTensor:  # type: ignore
+        """
+        Args:
+            boxes: source bounding boxes, Nx4 or Nx6 torch tensor or ndarray. The box mode is assumed to be ``StandardMode``
+            src_spatial_size: original image spatial size before zooming, used only when keep_size=True.
+        """
+        spatial_dims: int = get_spatial_dims(boxes=boxes)
+        _zoom = ensure_tuple_rep(self.zoom, spatial_dims)  # match the spatial image dim
+
+        if not self.keep_size:
+            return zoom_boxes(boxes, _zoom)
+
+        if src_spatial_size is None:
+            raise ValueError("keep_size=True, src_spatial_size must be provided.")
+
+        src_spatial_size = ensure_tuple_rep(src_spatial_size, spatial_dims)
+        dst_spatial_size = [int(round(src_spatial_size[axis] * _zoom[axis])) for axis in range(spatial_dims)]
+        zoomed_boxes = resize_boxes(boxes, src_spatial_size, dst_spatial_size)
+        if not np.allclose(np.array(src_spatial_size), np.array(dst_spatial_size)):
+            for axis, (od, zd) in enumerate(zip(src_spatial_size, dst_spatial_size)):
```

Comment:
I think maybe we should also put the `keep size` logic in the `box_ops`?
@wyli What do you think?

Thanks.

Reply:
yes, I think it's only useful in the context of the image zoom, perhaps we can refactor later when it's moved into monai.transforms https://github.com/Project-MONAI/MONAI/blob/a609bc62fb07a534d49ee8e7fe934c0001ff54a9/monai/transforms/spatial/array.py#L933-L947

Reply:
OKOK, that makes sense.

Thanks.